src/table.ts

```diff
@@ -937,19 +957,29 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
             return;
           }
           if (
-            numRequestsMade <= maxRetries &&
+            numConsecutiveErrors < maxRetries &&
             RETRYABLE_STATUS_CODES.has(error.code)
           ) {
-            makeNewRequest();
+            numConsecutiveErrors++;
+            const backOffSettings =
+              options.gaxOptions?.retry?.backoffSettings ||
+              DEFAULT_BACKOFF_SETTINGS;
+            const nextRetryDelay = getNextDelay(
+              numConsecutiveErrors,
+              backOffSettings
+            );
+            retryTimer = setTimeout(makeNewRequest, nextRetryDelay);
           } else {
             userStream.emit('error', error);
           }
         })
+        .on('data', _ => {
+          numConsecutiveErrors = 0;
```

Comment:
Please add a comment explaining that we reset retries when we get a successful result